tools/run-e2e-test.sh

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env bash
+
+# Copyright 2024 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Runs the e2e tests.
+#  Can be run in two modes: 1) command-line, and 2) prow CI.
+#  - Builds the e2e test binary (i.e. "test/e2e/e2e.test").
+#  - Packages e2e test binary with "ginkgo" and "kubectl" binaries.
+#  - Starts a cluster with kubetest2, using the "gce" deployer.
+#    - Cluster started in either passed cluster (GCP_CLUSTER env var),
+#      or "boskos" cluster in prow CI.
```

Comment:
TODO: revisit this versus letting kubetest2 boskos client handle this from go?

Reply:
> I don't think that works. We need to pass the GCE_PROJECT into the ginkgo-tester test args. But if we let kubetest2 create the boskos project, there doesn't seem to be a way to get the project name and then pass it into the test args. Please let me know if there is a way to do this.

IMHO the deployer should either set it as the current project (in gcloud?) and we depend on that in the test as the default or we should modify kubetest2 to support passing through information like this.

We could maybe smuggle it in the kubeconfig.

something TODO later, this is super unweildy and we should find a better answer.